charts/astronomer/templates/ingress.yaml

```diff
@@ -13,20 +13,38 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     heritage: {{ .Release.Service }}
   annotations:
-    {{- if .Values.global.authSidecar.enabled  }}
-{{- toYaml .Values.global.extraAnnotations | nindent 4 }}
+    {{- if .Values.global.authSidecar.enabled }}
+    {{- if .Values.global.extraAnnotations }}
+    {{- $filteredAnnotations := dict -}}
+    {{- range $key, $value := .Values.global.extraAnnotations -}}
+      {{- if ne $key "kubernetes.io/ingress.class" -}}
+        {{- $_ := set $filteredAnnotations $key $value -}}
+      {{- end -}}
+    {{- end -}}
+    {{- if $filteredAnnotations }}
+{{- toYaml $filteredAnnotations | nindent 4 }}
+    {{- end }}
+    {{- end }}
     {{- else }}
     {{- if .Values.global.extraAnnotations }}
-{{- toYaml .Values.global.extraAnnotations | nindent 4 }}
+    {{- $filteredAnnotations := dict -}}
+    {{- range $key, $value := .Values.global.extraAnnotations -}}
+      {{- if ne $key "kubernetes.io/ingress.class" -}}
+        {{- $_ := set $filteredAnnotations $key $value -}}
+      {{- end -}}
+    {{- end -}}
+    {{- if $filteredAnnotations }}
+{{- toYaml $filteredAnnotations | nindent 4 }}
+    {{- end }}
```

Comment:
I'd love it if we could find a way to put this long snip of code that is pasted repeatedly into the templates into something that is reusable, like a named template in _helpers.tpl